contrib/storage-hive/core/src/main/java/org/apache/drill/exec/planner/sql/logical/ConvertHiveParquetScanToDrillParquetScan.java

```diff
@@ -101,12 +101,24 @@ public boolean matches(RelOptRuleCall call) {
       return true;
     }
 
+    final List<FieldSchema> tableSchema = hiveTable.getSd().getCols();
     // Make sure all partitions have the same input format as the table input format
     for (HivePartition partition : partitions) {
-      Class<? extends InputFormat> inputFormat = getInputFormatFromSD(hiveTable, partition.getPartition().getSd());
+      final StorageDescriptor partitionSD = partition.getPartition().getSd();
+      Class<? extends InputFormat> inputFormat = getInputFormatFromSD(hiveTable, partitionSD);
       if (inputFormat == null || !inputFormat.equals(tableInputFormat)) {
         return false;
       }
+
+      // Make sure the schema of the table and schema of the partition matches. If not return false. Currently native
```

Comment:
Could you add a minor comment indicating that the schema changes between partition and table can happen due to "alter table" statements and that we would need the converter functions in case the type of the column has been changed via the alter table.